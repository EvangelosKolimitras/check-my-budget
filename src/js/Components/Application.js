import React from 'react'

export default class Application extends React.Component {
    render() {
        return(
           <div>
                <h2>Welcome to my MDA Boilerplate v1.0.0</h2>
                <h6> created by: <a href="http://e-kolimitras.com/"> Evangelos Kolimitras </a> </h6>
                <p>Contributions are mostly welcome and accepted</p>
           </div>
        )
    }
}
