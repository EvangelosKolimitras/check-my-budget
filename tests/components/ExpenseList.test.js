import React from "react"
import { ExpenseList } from "../../src/scripts/js/Components/ExpenseList"
import expenses from "../Fixtures/expenses"
import toJson from "enzyme-to-json"
import { shallow, configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
configure({ adapter: new Adapter() })

describe("Components", () => {
    describe("the ExpenseList component", () => {
        it("should render expenseList with expenses", () => {
            const wrapper = shallow(<ExpenseList expenses={expenses} />)
            console.log(wrapper)
            expect(toJson(wrapper)).toMatchSnapshot()
        })

        it("should render expenseList with no expenses", () => {
            const wrapper = shallow(<ExpenseList expenses={[]} />)
            console.log(wrapper)
            expect(toJson(wrapper)).toMatchSnapshot()
        })
    })
})
