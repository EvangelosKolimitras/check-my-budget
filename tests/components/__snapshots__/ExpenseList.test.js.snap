// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components the ExpenseList component should render expenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={3}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={2}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={1}
    createdAt={345600000}
    description="Credit"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`Components the ExpenseList component should render expenseList with no expenses 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
